### PR TITLE
feat: add use case to check current client id

### PR DESCRIPTION
**Pull request by saleniuk**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Currently there is no way to get the currently registered client id for the user or even to check if it has been already registered. We will need that option to check at the app launch if we need to open the "register client" screen.

### Solutions

Implemented CurrentClientIdUseCase and added to ClientScope.

### Testing

Added CurrentClientIdUseCaseTest.
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ClientId
+import com.wire.kalium.logic.functional.suspending
+
+interface CurrentClientIdUseCase {
+    suspend operator fun invoke(): CurrentClientIdResult
+}
+
+class CurrentClientIdUseCaseImpl(private val clientRepository: ClientRepository) : CurrentClientIdUseCase {
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt:11]: Do you actually need the clientID? My first though is that the consumer shouldn't deal with that and we can instead have a `NeedsToRegisterClientUseCase` which returns a boolean.

**Review by typfel — CHANGES_REQUESTED** [review]

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt (hunk 1 above)
Comment (saleniuk) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt:11]: That's a good point, but we already have `SelfClientsUseCase` so the consumer can already get even more than just `ClientId` 🤔 @typfel let me know if I should still change it to `NeedsToRegisterClientUseCase` 😉

Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt:11]: Well depends, do you intend to use the client ID where you use the use case or will you just throw it away and look for the error? If it's the latter I think it cleaner and more readable to have a `NeedsToRegisterClientUseCase`.

Comment (saleniuk) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/CurrentClientIdUseCase.kt]: You're right, I changed it to `NeedsToRegisterClientUseCase` which returns a boolean.

**Review by typfel — APPROVED** [review]

**Pull request merged**